### PR TITLE
Major update to boolean planning, adds simplify, etc.

### PR DESCRIPTION
**Pull request by pentacular**

### CODE REVIEW

File: nb/projects/micro_gear_motor/micro_gear_motor.md [hunk 1]
```diff
@@ -172,9 +172,9 @@ Motor Driver Holder
 
 ```JavaScript
 const wemosDriverHolder = Block(4, 5, 3.2 * 11)
-  .cut({ mode: 'basic' }, Box(25.7 + 0.2, 34.5 + 0.2).ez(2, 100))
```
Comment (BarbourSmith) [inline nb/projects/micro_gear_motor/micro_gear_motor.md:175]: Are we operating in a single mode now?

File: nb/projects/plotter/mount.nb [hunk 2]
```diff
@@ -79,6 +83,8 @@ const toothedReel = Group(
   .as('toothedReel')
   .view(rx(1 / 2));
 
+/*
+// FIX: These are too slow.
```
Comment (BarbourSmith) [inline nb/projects/plotter/mount.nb:87]: We should use the wasm thread support to compute our threads faster 😛

**Review by BarbourSmith — APPROVED** [review]
Amazing, stoked to try this. I should be able to update and put some miles on it tomorrow if this is merged.

File: api/shape/drop.js [hunk 3]
```diff
@@ -1,38 +1,7 @@
...
-            return Shape.fromGeometry(geometry).void().toGeometry();
-          } else if (geometry.type !== 'item') {
-            return descend();
-          }
-        }
-      }
-    };
-    return Shape.fromGeometry(rewrite(shape.toGeometry(), op));
-  };
+export const drop = (tag) => (shape) => shape.on(get(tag), voidFn());
```
[the remark this answers is not in the document]
Comment (pentacular) [reply api/shape/drop.js:5]: More or less we generalized what drop and keep were doing and produced better primitives.

This just rebases them on their children. :)

File: api/shape/sx.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import Shape from './Shape.js';
+import { scale } from './scale.js';
+
+export const scaleX =
```
[the remark this answers is not in the document]
Comment (pentacular) [reply api/shape/sx.js:4]: Mostly for mirroring.

I found I was writing a.and(scale(1, -1, 1)) a lot.

So I replaced that with a.sy(1, -1), etc.

File: nb/projects/micro_gear_motor/micro_gear_motor.md (hunk 1 above)
Comment (pentacular) [reply nb/projects/micro_gear_motor/micro_gear_motor.md:175]: Yeah, we might need this back, but let's finish consolidating first.

File: nb/projects/plotter/mount.nb (hunk 2 above)
Comment (pentacular) [reply nb/projects/plotter/mount.nb:87]: Yeah, although I think this is a different problem.

I'll need to come back and figure out what's happening here.

Comment (BarbourSmith) [reply nb/projects/plotter/mount.nb:87]: Yeah, it wasn't a particularly insightful comment. I just wanted to make the joke 😁

**Pull request merged**